lib/core/src/rcMisc.cpp

```diff
@@ -4684,7 +4684,8 @@ int get_canonical_name(const char *_hostname, char* _buf, size_t _len)
     snprintf(_buf, _len, "%s", p_addrinfo->ai_canonname);
 
     if (CLIENT_PT != ::ProcessType) {
-        const std::chrono::seconds age{irods::get_dns_cache_eviction_age()};
+        const auto cache_config = irods::get_advanced_setting<nlohmann::json>(irods::KW_CFG_DNS_CACHE);
+        const std::chrono::seconds age{cache_config.at(irods::KW_CFG_EVICTION_AGE_IN_SECONDS).get<int>()};
```

Comment:
I realize string allocations can throw and we are leaning on the schema validation, but I think we should handle exceptions around this.

Reply:
Done.